pilot/pkg/model/service.go

```diff
@@ -822,7 +829,17 @@ func IsValidSubsetKey(s string) bool {
 func ParseSubsetKey(s string) (direction TrafficDirection, subsetName string, hostname Hostname, port int) {
 	parts := strings.Split(s, "|")
 	if len(parts) < 4 {
-		return
+		// This could be the DNS srv form of the cluster that uses outbound_.port_.subset_.hostname
+		// Since we dont want every callsite to implement the logic to differentiate between the two forms
+		// we add an alternate parser here.
+		if !strings.HasPrefix(s, fmt.Sprintf("%s_", TrafficDirectionOutbound)) &&
+			!strings.HasPrefix(s, fmt.Sprintf("%s_", TrafficDirectionInbound)) {
+			return
+		}
+		parts = strings.SplitN(s, ".", 4)
```

Comment:
It's safer to do Split(s, ".") and check the number of parts. The above SplitN() call doesn't preclude more then 4 parts.

Reply:
we only need 4 parts. Also, this is not coming from end user. This is generated by cluster.go. So we know that the cluster name has 4 parts definitely